### PR TITLE
[CMAKE/WIDL] Add rudimentary dependency tracking in add_rpcproxy_files

### PR DESCRIPTION
**Pull request by learn-more**
JIRA issue: [CORE-14204](https://jira.reactos.org/browse/CORE-14204)

### CODE REVIEW

File: sdk/cmake/widl-support.cmake [hunk 1]
```diff
@@ -45,11 +45,20 @@ function(add_rpcproxy_files)
 
     foreach(FILE ${ARGN})
         get_filename_component(NAME ${FILE} NAME_WE)
-            list(APPEND IDLS ${CMAKE_CURRENT_SOURCE_DIR}/${FILE})
-            add_custom_command(
-                OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/${NAME}_p.c ${CMAKE_CURRENT_BINARY_DIR}/${NAME}_p.h
-                COMMAND native-widl ${INCLUDES} ${DEFINES} ${IDL_FLAGS} -p -o ${CMAKE_CURRENT_BINARY_DIR}/${NAME}_p.c -h -H ${NAME}_p.h ${CMAKE_CURRENT_SOURCE_DIR}/${FILE}
-                DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/${FILE} native-widl)
+        string(REPLACE "_" ";" SPLIT_FILE ${FILE})
```
Comment (ThFabba) [inline sdk/cmake/widl-support.cmake]: Perhaps add a little comment explaining the file name format that this heuristic is based on?

Comment (learn-more) [reply sdk/cmake/widl-support.cmake]: Good idea.

File: sdk/cmake/widl-support.cmake [hunk 2]
```diff
@@ -45,11 +45,22 @@ function(add_rpcproxy_files)
...
-            list(APPEND IDLS ${CMAKE_CURRENT_SOURCE_DIR}/${FILE})
-            add_custom_command(
-                OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/${NAME}_p.c ${CMAKE_CURRENT_BINARY_DIR}/${NAME}_p.h
-                COMMAND native-widl ${INCLUDES} ${DEFINES} ${IDL_FLAGS} -p -o ${CMAKE_CURRENT_BINARY_DIR}/${NAME}_p.c -h -H ${NAME}_p.h ${CMAKE_CURRENT_SOURCE_DIR}/${FILE}
-                DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/${FILE} native-widl)
+        # Most proxy idl's have names like <proxyname>_<original>.idl
+        # We use this to create a dependency from the proxy to the original idl
+        string(REPLACE "_" ";" SPLIT_FILE ${FILE})
+        list(LENGTH SPLIT_FILE len)
+        set(EXTRA_DEP )
```
Comment (HBelusca) [inline sdk/cmake/widl-support.cmake]: Extra space before the closing parenthesis?

Comment (learn-more) [reply sdk/cmake/widl-support.cmake]: Intentional.

**Pull request merged**